### PR TITLE
Add support for actionable notifications on Wear OS directly

### PR DESCRIPTION
**Pull request by dshokouhi**
## Summary

Fixes: #3324 by adding support for actionable notifications.

A few things to note:

- Given the dependency on `messagingManager` and that we need variables initialized from the receiver I was unable to move `NotficationActionReceiver` as is so have opted to copy it over.
- `URI` action type is not supported at the moment, only reply and actions.
- Updating reply history on my Pixel Watch results in a weird notification, almost like a bug in how the pixel watch displays things. Same code works fine on the phone so feels like a platform difference.

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#908

### CODE REVIEW

**Comment by jpelgrom** [comment]
That's quite a lot of duplicated code... Did you consider making a shared base class like we do for the sensor worker, with the `app`/`wear` `MessagingManager` exposed using an abstract value/function?

**Comment by dshokouhi** [comment]
> That's quite a lot of duplicated code...

I know, not a fan but I also have not come up with a good and viable solution. There are too many dependencies on things in the `app` module that causes a conflict trying to migrate to `common`.

File: common/src/main/java/io/homeassistant/companion/android/common/notifications/NotificationFunctions.kt [hunk 1]
```diff
@@ -274,3 +290,101 @@ fun handleText(
...
+        action.build()
+    )
+        .build()
+}
+
+fun createPendingIntent(
+    context: Context,
+    requestCode: Int,
+    intent: Intent,
+    type: String
```
Comment (jpelgrom) [inline common/src/main/java/io/homeassistant/companion/android/common/notifications/NotificationFunctions.kt:326]: As this is only used as a flag to check if it is a reply or not a boolean would be better. If you have future plans use an enum or constants :)

**Review by jpelgrom — COMMENTED** [review]
Considering the duplicated receiver again: if `MessagingManager` is the only app/wear specific dependency, couldn't we move it to the common module with an abstract function to be implemented in the app/wear module that uses the `MessagingManager`? By doing it like that the common module doesn't actually know about the messaging, and the only duplicated code is ~8 lines instead of the entire class.

**Comment by dshokouhi** [comment]
leaving as a draft while I look into using abstract functions

**Comment by dshokouhi** [comment]
Closing as I currently do not have enough free time to look into the requested changes

**Pull request closed without merging**